src/internal/Observable.ts

```diff
@@ -241,11 +248,16 @@ export class Observable<T> implements Subscribable<T> {
       if (config.useDeprecatedSynchronousErrorHandling) {
         sink.syncErrorThrown = true;
         sink.syncErrorValue = err;
-      }
-      if (canReportError(sink)) {
-        sink.error(err);
       } else {
-        console.warn(err);
+        if (canReportError(sink)) {
+          sink.error(err);
+        } else {
+          // This is hit when the user develops a poorly made observable. That means that
+          // something is calling `subscriber.error` more than once. We're currently logging
+          // this as a warning to console to let them know something is wrong.
```

Comment:
issue: This comment is not quite correct, AFAICT. Execution can only arrive here if an error is thrown - not emitted as notifications via the observer's `error` method. Specifically, the `canReportError` business was added because implementation errors - within the RxJS core - were being swallowed when unintentional errors where thrown after `complete` or `error` notifications had been made. And the swallowed errors meant the problems were hard-to-find.

IMO, we should edit the comment to reflect this. I feel strongly that - with the upcoming major version change - we should take the opportunity to simplify this (in a separate PR). I am strongly of the opinion that calls to an observer's `error` method should effect a call to `hostReportError` - or some developer-configurable hook - rather than do what's done now if the subscriber is stopped: nothing (see [here](https://github.com/ReactiveX/rxjs/blob/15f13e961d66dbb4774c06ebea7da586f47990ed/src/internal/Subscriber.ts#L106-L111) and [here](https://github.com/ReactiveX/rxjs/blob/15f13e961d66dbb4774c06ebea7da586f47990ed/src/internal/Subscriber.ts#L201-L229)). I am of the opinion that there is no greater sin than swallowing errors - especially within a library - and if there is any hill upon which I would be prepared to die, this would be it.

Reply:
@cartant. I have a follow-up PR here that adds a new configurable unhandled error hook, and removes this `console.warn`:

https://github.com/ReactiveX/rxjs/pull/5681